src/ngRoute/route.js

```diff
@@ -160,7 +171,7 @@ function $RouteProvider() {
    */
   this.when = function(path, route) {
     //copy original route object to preserve params inherited from proto chain
-    var routeCopy = angular.copy(route);
+    var routeCopy = createShallowCopy(route);
```

Comment:
Since we are in a external module, we need to copy it. I started by copying that, but then removed the stuff that didn't seem relevant.